### PR TITLE
GitHub audit aws

### PR DESCRIPTION
**Pull request by gibbardsteve**
Add terraform configuration for deployment to either sandbox, dev or prod.

Terraform is divided into three separate modules, each of which is run separately

Authentication - Cognito authentication user pool config
Storage - S3 stroage used by the application to store data
Service - The actual github-audit service that launches the container

The reason to have three separate Terraform configs is to allow the service to be applied and destroyed independantly without needing to reconfigure users or lose data.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -3,5 +3,7 @@
 venv
 repositories.json
 archived.json
+userlist.json
```
Comment (TotalDwarf03) [inline .dockerignore:6]: Not required anymore

File: README.md [hunk 2]
```diff
@@ -155,30 +159,157 @@ You can find the AWS repo push commands under your repository in ECR by selectin
...
+
+- Apply the changes, ensuring you use the correct environment config (depending upon which env you are configuring):
+
+  E.g. for the dev environment run
+
+  ```bash
+  terraform apply -var-file=env/dev/dev.tfvars
+  ```
+
+- When the terraform has applied successfully the running task will have been replaced by a task running the container version you specified in the tfvars file
```
Comment (TotalDwarf03) [inline README.md:313]: Move the content to its own section. (or delete it as it won't be needed in a future version)

File: repoarchivetool/app.py [hunk 3]
```diff
@@ -9,11 +9,17 @@
 import authentication_interface
 
 archive_threshold_days = 30
-organisation = "ONS-Innovation"
 
 app = flask.Flask(__name__)
 app.config['SECRET_KEY'] = os.urandom(24)
 
+# Set variables from the environment
+account = os.getenv("AWS_ACCOUNT_NAME")
```
Comment (TotalDwarf03) [inline repoarchivetool/app.py:17]: Where these are pulled from the running environment, won't instructions for running outside a container be different? The user would have to provide a .env file and that'll have to be loaded into the environment.

Might be easier to remove local running instructions and force running in a container only?

**Review by TotalDwarf03 — COMMENTED** [review]
Added a couple of minor things - mostly picked up when we ran through them.

There is a comment about how the environment variables get initialised when running the application outside of a container. Free to discuss/further test this

File: repoarchivetool/app.py (hunk 3 above)
Comment (gibbardsteve) [reply repoarchivetool/app.py:17]: os.getenv should just pickup the value if you have configured it locally before running the app
i.e export AWS_ACCOUNT_NAME=xyz

Do you want me to add something to the readme?

Comment (gibbardsteve) [reply repoarchivetool/app.py:17]: Added to README

**Review by TotalDwarf03 — APPROVED** [review]
Corrections completed, ready to merge

**Pull request merged**